### ISSUE TITLE
Lowercase acgt letters in fasta files are regarded as N by catwalk

### ISSUE THREAD

**Issue by davidhwyllie** [dvolk/catwalk#10]
It appears that lower case acgt letters in fasta files are regarded as N by catwalk, or perhaps mishandled in some complex way.
This applies both to the _mfsl endpoint and to the /add_sample endpoint.

Historically no standard has applied for upper and lower case letters in fasta files.
However, a commonly used convention uses lower case to indicate areas of lower quality, but lower case is sometimes used for other reasons.  For example, the fasta files containing covid simulation data (see benchmark/sim) are in lower case, and this would not normally be regarded as 'wrong' in the molecular biology community.  Therefore, ignoring acgt characters is not appropriate behaviour.

What should happen:
a,c,g,t should be handled in the same way as A,C,G,T, respectively.

What happens now:
ac,g,t are regarded as missing data (N) or maybe mishandled in some more complex way (see below)

Impact:
This causes incorrect results to be generated in a manner which would not be predicted by the user.

How to test/ unit test:
Add the below to test/test.py below line 86.  This adds a variant of test sequence test2, with a lower case rather than upper case character as its last character.

 ```
    r = add_sample_to_catwalk("test2l", "ACGTACGg")   # is lower case handled? ref is ACGTACGT; should give the same result as test2
    n2l = get_neighbours("test2l", 50)
    print(">>> n2", n2)
    print(">>>n2l", n2l)
    assert n2l == [["test3", "1"], ["test1", "1"],  ["test2", "0"]]
    assert r.status_code == 201
```

Current response:

>>> n2 [['test3', '1'], ['test1', '1']]
>>>n2l [['test1', '0'], ['test2', '0']]

test1 = ACGTACGT  # exactly matches this (wrong)
test2 = ACGTACGG # exactly matches this (wrong)
test3 = ACGTACGC # doesn't match this at 1 snp (also wrong)
test4 = ACGTACGC # doesn't match match this at 1 snp (also wrong)
test2l   ACGTACGg

**Comment by dvolk** [comment]
Thanks for reporting. I've added handling for samples having lowercase bases.

Can the reference have lowercase bases too?

**Comment by davidhwyllie** [comment]
Genbank references/fasta files don't.   However, it would be wise to upper case the reference when it is added, if this is relevant.
Thank you for looking into this.

**Comment by dvolk** [comment]
When you're adding samples with /add_sample_from_refcomp, do you expect to pass in keys with lower case bases, i.e. {'a': ..., 'c': ...} or is it always upper case?

**Comment by davidhwyllie** [comment]
No, I think we can stipulate that the keys have to be upper case for this endpoint.  It just has to be carefully documented.

**Comment by dvolk** [comment]
Fixed in 0.9.3
